projects/ngrx.io/content/guide/component-store/usage.md

```diff
@@ -0,0 +1,239 @@
+
+Finally, the state is aggregated with selectors into two properties:
+* `vm$` property collects all the data needed for the template - this is *ViewModel* of `SlideToggleComponent`. 
+* `change` is the `@Output` of `SlideToggleComponent`. Instead of creating `EventEmitter` and push the data through it, here the output is connected to the Observable source directly.
+
+<code-example
+  path="component-store-slide-toggle/src/app/slide-toggle.component.ts"
+  region="selector"></code-example>
+
+This example does not have a lot of business logic, however it is still fully reactive.
+
+#### Zoneless
+
+In the template of `SlideToggleComponent` we used `async` pipe to unwrap the Observable, however in the Zoneless environment it will not work. In such cases, [ngrxPush pipe](guide/component/push) can be used instead. 
+
+
+### Example 2: Service extending ComponentStore
+
+`SlideToggleComponent` is a fairly simple component and having `ComponentStore` within the component itself is still manageable. When components takes more Inputs and/or has more events within its template, it becomes larger and harder to read/maintain.
+
+Extracting the business logic of a component into a separate Service also helps reduce the cognitive load while reading the components code.
+
+<div class="alert is-important">
+
+A Service that extends ComponentStore and contains business logic of the component brings many advantages. **This is also the recommended usage of ComponentStore**.
+
+</div>
+
+`ComponentStore` was designed with such approach in mind. The main APIs of `ComponentStore` (`updater`, `effect` and `select`) are meant to wrap the **HOW** state is changed, extracted or effected, and then provided with inputs.
+
+Below are the two examples of re-implemented [Paginator component](https://material.angular.io/components/paginator/overview) from Angular Material library of UI components. These re-implementations are very functionally close alternatives.
+
+Here's the source code of the [Material's paginator.ts](https://github.com/angular/components/blob/23d3c216c65b327e0acfb48b53302b8fda326e7f/src/material/paginator/paginator.ts#L112) for the reference.
+
+What we can see is that while the *"PaginatorComponent providing ComponentStore"* example already makes the component a lot smaller, reactive, removes `this._changeDetectorRef.markForCheck()` and organizes it into distinct "read"/"write"/"effect" buckets, it still could be harder to read. *"PaginatorComponent with PaginatorStore"* example adds readability and further improves testability of behaviors and business logic.
+
+<div class="alert is-helpful">
+
+You can see the examples at StackBlitz:
+* "PaginatorComponent providing ComponentStore" <live-example name="component-store-paginator" noDownload></live-example>
+* "PaginatorComponent with PaginatorStore Service" <live-example name="component-store-paginator-service" noDownload></live-example>
+
+</div>
+
+<code-tabs linenums="true">
```

Comment:
Nit, I find this codeblock too big to be added to the docs.
It also makes it hard to scroll to the bottom of the page.

Reply:
I want to make it easier to compare these two side-by-side. I'll extract the init state into PaginatorStore - that should make it a bit smaller. (was planning to do it anyway :) )

Reply:
Thanks Alex!